numpy/f2py/crackfortran.py

```diff
@@ -2892,17 +2882,16 @@ def compute_deps(v, deps):
                                 vars[v]['depend'] = list(set(v_deps))
                             if n not in v_deps:
                                 n_deps.append(v)
-            elif isstring(vars[n]):
-                if 'charselector' in vars[n]:
-                    if '*' in vars[n]['charselector']:
-                        length = _eval_length(vars[n]['charselector']['*'],
-                                              params)
-                        vars[n]['charselector']['*'] = length
-                    elif 'len' in vars[n]['charselector']:
-                        length = _eval_length(vars[n]['charselector']['len'],
-                                              params)
-                        del vars[n]['charselector']['len']
-                        vars[n]['charselector']['*'] = length
+            elif isstring(vars[n]) and 'charselector' in vars[n]:
+                if '*' in vars[n]['charselector']:
+                    length = _eval_length(vars[n]['charselector']['*'],
+                                          params)
```

Comment:
No need to break this line.

Reply:
I've just run ruff with this specific rule automatically. A formatter would unfold the line, but the linter doesn't.